stl/inc/algorithm

```diff
@@ -160,6 +165,38 @@ _Ty* __std_find_last_trivial(_Ty* const _First, _Ty* const _Last, const _TVal _V
         static_assert(_Always_false<_Ty>, "Unexpected size");
     }
 }
+
+template <class _Ty>
+_Ty* __std_find_first_of_trivial(
+    _Ty* const _First1, _Ty* const _Last1, _Ty* const _First2, _Ty* const _Last2) noexcept {

```

Comment:
This breaks existing code with varying constness, which also indicates missing test coverage:

```
D:\GitHub\STL\out\x64>type meow.cpp
```
```cpp
#include <algorithm>
#include <cassert>
#include <vector>
using namespace std;

int main() {
    const vector<char> haystack{'m', 'e', 'o', 'w', 'C', 'A', 'T', 'S'};
    vector<char> needle{'R', 'S', 'T'};
    const auto result = find_first_of(haystack.begin(), haystack.end(), needle.begin(), needle.end());
    assert(result == haystack.begin() + 6);
}
```
```
D:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od /D_USE_STD_VECTOR_ALGORITHMS=0 meow.cpp && meow
meow.cpp

D:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od meow.cpp && meow
meow.cpp
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): error C2672: 'std::__std_find_first_of_trivial': no matching overloaded function found
D:\GitHub\STL\out\x64\out\inc\algorithm(170): note: could be '_Ty *std::__std_find_first_of_trivial(_Ty *const ,_Ty *const ,_Ty *const ,_Ty *const ) noexcept'
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): note: '_Ty *std::__std_find_first_of_trivial(_Ty *const ,_Ty *const ,_Ty *const ,_Ty *const ) noexcept': template parameter '_Ty' is ambiguous
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): note: could be '_Ty'
        with
        [
            _Ty=char
        ]
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): note: or       '_Ty'
        with
        [
            _Ty=const char
        ]
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): note: '_Ty *std::__std_find_first_of_trivial(_Ty *const ,_Ty *const ,_Ty *const ,_Ty *const ) noexcept': could not deduce template argument for '_Ty *const ' from '_Ty *'
        with
        [
            _Ty=char
        ]
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): note: the template instantiation context (the oldest one first) is
meow.cpp(9): note: see reference to function template instantiation '_FwdIt1 std::find_first_of<std::_Vector_const_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>>(const _FwdIt1,const _FwdIt1,const _FwdIt2,const _FwdIt2)' being compiled
        with
        [
            _FwdIt1=std::_Vector_const_iterator<std::_Vector_val<std::_Simple_types<char>>>,
            _Ty=char,
            _FwdIt2=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<char>>>
        ]
D:\GitHub\STL\out\x64\out\inc\algorithm(3397): note: see reference to function template instantiation '_FwdIt1 std::find_first_of<_FwdIt1,_FwdIt2,std::equal_to<void>>(_FwdIt1,const _FwdIt1,const _FwdIt2,const _FwdIt2,_Pr)' being compiled
        with
        [
            _FwdIt1=std::_Vector_const_iterator<std::_Vector_val<std::_Simple_types<char>>>,
            _FwdIt2=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<char>>>,
            _Pr=std::equal_to<void>
        ]
D:\GitHub\STL\out\x64\out\inc\algorithm(3367): error C2737: '_Result': const object must be initialized
D:\GitHub\STL\out\x64\out\inc\algorithm(3371): error C3536: '_Result': cannot be used before it is initialized
D:\GitHub\STL\out\x64\out\inc\algorithm(3371): error C2440: '=': cannot convert from 'int' to 'const char *'
D:\GitHub\STL\out\x64\out\inc\algorithm(3371): note: Conversion from integral type to pointer type requires reinterpret_cast, C-style cast or parenthesized function-style cast
```